test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/PropertyTrackingTests/ChangeTrackingOptionsTests.cs

```diff
@@ -0,0 +1,720 @@
+        Assert.Equal(datetime, people0.Orders.First().OrderDate);
+        Assert.Equal(2, people0.Orders.First().OrderShelfLifes.Count());
+
+        context.LoadProperty(accounts[0], "MyGiftCard");
+
+        //  Update single value navigation property .
+        Assert.NotNull(accounts[0].MyGiftCard);
+        accounts[0].MyGiftCard.ExperationDate = datetime;
+        Action<WritingEntryArgs> onEntryEndingAction3 = (arg) =>
+        {
+            if (arg.Entry.TypeName.EndsWith("GiftCard"))
+            {
+                Assert.Equal(expectedPropertyCount, arg.Entry.Properties.Count());
+            }
+        };
+        context.Configurations.RequestPipeline.OnEntryEnding(onEntryEndingAction3);
+        expectedPropertyCount = 1;
+        context.SaveChanges();
+
+        // Update open complex type
+        accounts[0].AccountInfo.LastName = "S.";
+        Action<WritingEntryArgs> onEntryEndingAction4 = (arg) =>
+        {
+            if (arg.Entry.TypeName.EndsWith("AccountInfo"))
+            {
+                Assert.Equal(expectedPropertyCount, arg.Entry.Properties.Count());
+                Assert.Equal("S.", (arg.Entry.Properties.Single(p => p.Name == "LastName") as ODataProperty)?.Value);
+            }
+        };
+        context.Configurations.RequestPipeline.OnEntryEnding(onEntryEndingAction4);
+
+        expectedPropertyCount = 3;
+        context.SaveChanges();
+
+        var account = context.Accounts.Expand("MyGiftCard").Where((it) => it.AccountID == accounts[0].AccountID).Single();
+        Assert.Equal(datetime, account.MyGiftCard.ExperationDate);
+        Assert.Equal("S.", account.AccountInfo.LastName);
+        Assert.Equal("New", account.AccountInfo.FirstName);
+
+        // Update property in open singleton
+        publicCompany.Single().Revenue = 200;
+        publicCompany.Single().Name = "MS Ltd.";
+        Action<WritingEntryArgs> onEntryEndingAction5 = (arg) =>
+        {
+            if (arg.Entry.TypeName.EndsWith("Company"))
+            {
+                Assert.Equal(expectedPropertyCount, arg.Entry.Properties.Count());
+            }
+        };
+        context.Configurations.RequestPipeline.OnEntryEnding(onEntryEndingAction5);
+        expectedPropertyCount = 2;
+        context.SaveChanges();
+
+        var company = context.PublicCompany.GetValue();
+        Assert.Equal(200, company.Revenue);
+        Assert.Equal("MS Ltd.", company.Name);
+
+        // Update object by update object without change => redo the PATCH all
+        context.UpdateObject(people[0]);
+
+        expectedPropertyCount = 10;
+        context.SaveChanges();
+    }
+
+    // This test verifies that updating an entity without any changes results in a PATCH request
+    // that includes all properties of the entity. It ensures that the request pipeline correctly
+    // handles such scenarios and includes the expected number of properties in the request.
+    [Fact]
+    public void UpdateEntityWithoutChanges()
+    {

```

Comment:
This too. Rename the test name well in a way that is easy to understand what the test is doing.

Reply:
Is this when using a `DataServiceCollection` or even when it's not used.